### PR TITLE
support is added for srv6,srv6_te,srv6_isis templates

### PR DESCRIPTION
**Pull request by swarudka**
All test are running successfully, except segment_routing_candidate_paths (Cleanup is not happening, can you please suggest here)

### CODE REVIEW

File: gen/definitions/segment_routing_v6.yaml [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+    attributes:
+      - yang_name: locator-enable
+        example: true
+      - yang_name: name
+        id: true
+        example: Locator1        
+      - yang_name: micro-segment/behavior
+        example: unode-psp-usd
+      - yang_name: prefix/prefix
+        tf_name: locator_prefix
```
Comment (danischm) [inline gen/definitions/segment_routing_v6.yaml:22]: As this is already part of the `locators` list the tf_name should just be `prefix`.

File: gen/definitions/segment_routing_v6.yaml [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+        example: "fccc:0:214::"
+      - yang_name: prefix/prefix-length
```
Comment (danischm) [inline gen/definitions/segment_routing_v6.yaml:24]: We should add a tf_name with `prefix_length`.

File: gen/definitions/segment_routing_isis.yaml [hunk 3]
```diff
@@ -0,0 +1,26 @@
+---
```
Comment (danischm) [inline gen/definitions/segment_routing_isis.yaml:1]: We already have a resource `iosxr_router_isis_address_family` so we should add the missing attributes there.

File: gen/definitions/segment_routing_te.yaml [hunk 4]
```diff
@@ -0,0 +1,64 @@
+---
+name: Segment Routing Traffic Engineering
```
Comment (danischm) [inline gen/definitions/segment_routing_te.yaml:2]: We should change the name to `Segment Routing TE` to be consistent.

File: gen/definitions/segment_routing_te.yaml [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    tf_name: on_demand_colors
+    type: List
+    attributes: 
+      - yang_name: color
+        id: true
+        example: 266
+      - yang_name: srv6/enable  
+        example: true
+        exclude_test: true
+      - yang_name: srv6/locator/locator-name
```
Comment (danischm) [inline gen/definitions/segment_routing_te.yaml:16]: tf_name: `srv6_locator_name`

File: gen/definitions/segment_routing_te.yaml [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+        example: LOC11
+      - yang_name: srv6/locator/behavior
+        example: ub6-insert-reduced
+      - yang_name: srv6/locator/binding-sid-type
+        example: srv6-dynamic
+      - yang_name: source-address/source-address
```
Comment (danischm) [inline gen/definitions/segment_routing_te.yaml:22]: tf_name: `source_address`

File: gen/definitions/segment_routing_te.yaml [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+        example: fccc:0:213::1
+      - yang_name: source-address/ip-address-type
```
Comment (danischm) [inline gen/definitions/segment_routing_te.yaml:24]: tf_name: `source_address_type`

File: gen/definitions/segment_routing_te.yaml [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+        example: end-point-type-ipv6
+      - yang_name: effective-metric/enable
+        example: true
+        exclude_test: true
+      - yang_name: effective-metric/metric-value-type/metric-value
```
Comment (danischm) [inline gen/definitions/segment_routing_te.yaml:29]: tf_name: `effective_metric_value`

File: gen/definitions/segment_routing_te.yaml [hunk 9]
```diff
@@ -0,0 +1,64 @@
...
+        example: 4444
+      - yang_name: effective-metric/metric-value-type/metric-type
```
Comment (danischm) [inline gen/definitions/segment_routing_te.yaml:31]: tf_name: `effective_metric_type`

File: gen/definitions/segment_routing_te.yaml [hunk 10]
```diff
@@ -0,0 +1,64 @@
...
+    tf_name: policies
+    type: List
+    attributes:
+      - yang_name: policy-name
+        id: true
+        example: POLICY1
+      - yang_name: srv6/enable
+        example: true
+        exclude_test: true
+      - yang_name: srv6/locator/locator-name
```
Comment (danischm) [inline gen/definitions/segment_routing_te.yaml:49]: tf_name: `srv6_locator_name`

File: gen/definitions/segment_routing_te.yaml [hunk 11]
```diff
@@ -0,0 +1,64 @@
...
+        example: Locator11
+      - yang_name: srv6/locator/binding-sid-type
+        example: srv6-dynamic
+      - yang_name: srv6/locator/behavior
+        example: ub6-insert-reduced
+      - yang_name: source-address/source-address
```
Comment (danischm) [inline gen/definitions/segment_routing_te.yaml:55]: tf_name: `source_address`

File: gen/definitions/router_isis_address_family.yaml [hunk 12]
```diff
@@ -105,3 +105,21 @@ attributes:
...
+    type: List
+    attributes:
+      - yang_name: instance-id
+        id: true
+        example: CORE
+      - yang_name: route-policy
+        example: ROUTE_POLICY_1
+
+
+# test_prerequisites:
```
Comment (danischm) [inline gen/definitions/router_isis_address_family.yaml:120]: If it is not needed, this can be removed.

File: gen/definitions/l2vpn_xconnect_group_p2p.yaml [hunk 13]
```diff
@@ -19,6 +19,7 @@ attributes:
   - yang_name: neighbor/ipv4s/ipv4
     tf_name: ipv4_neighbors
     type: List
+    exclude_test: true
```
Comment (danischm) [inline gen/definitions/l2vpn_xconnect_group_p2p.yaml:22]: Why did you add `exclude_test` here?

File: gen/definitions/l2vpn_bridge_group_bridge_domain.yaml [hunk 14]
```diff
@@ -53,3 +53,5 @@ attributes:
         example: Bundle-Ether11.1234
       - yang_name: split-horizon/group
         example: true
+  - yang_name: segment-routing-srv6-evis/evi/vpn-id
```
Comment (danischm) [inline gen/definitions/l2vpn_bridge_group_bridge_domain.yaml:56]: tf_name: `segment_routing_srv6_evi`

File: gen/definitions/evpn_segment_routing_service.yaml [hunk 15]
```diff
@@ -0,0 +1,81 @@
+---
+name: EVPN Segment Routing Services
```
Comment (danischm) [inline gen/definitions/evpn_segment_routing_service.yaml:2]: name: `EVPN Segment Routing SRv6 EVI`

File: gen/definitions/evpn_segment_routing_service.yaml [hunk 16]
```diff
@@ -0,0 +1,81 @@
...
+    attributes:
+      - yang_name: two-byte-as-number
+        tf_name: as_number
+        id: true
+        example: 1
+      - yang_name: assigned-number
+        id: true
+        example: 1
+  - yang_name: bgp/route-target/import/four-byte-as-rts/four-byte-as-rt
+    tf_name: evpn_bgp_route_target_import_four_byte_as_format
```
Comment (danischm) [inline gen/definitions/evpn_segment_routing_service.yaml:22]: Let's remove the `evpn_` prefix here.

File: gen/definitions/evpn_segment_routing_service.yaml [hunk 17]
```diff
@@ -0,0 +1,81 @@
...
+    attributes:
+      - yang_name: four-byte-as-number
+        tf_name: as_number
+        id: true
+        example: 65536
+      - yang_name: assigned-number
+        id: true
+        example: 1
+  - yang_name: bgp/route-target/import/ipv4-address-rts/ipv4-address-rt
+    tf_name: evpn_bgp_route_target_import_ipv4_address_format
```
Comment (danischm) [inline gen/definitions/evpn_segment_routing_service.yaml:34]: Let's remove the `evpn_` prefix here.

File: gen/definitions/evpn_segment_routing_service.yaml [hunk 18]
```diff
@@ -0,0 +1,81 @@
...
+    exclude_test: true
+    attributes:
+      - yang_name: ipv4-address
+        id: true
+        example: 1.1.1.1
+      - yang_name: assigned-number
+        id: true
+        example: 1
+  - yang_name: bgp/route-target/export/two-byte-as-rts/two-byte-as-rt
+    tf_name: evpn_bgp_route_target_export_two_byte_as_format
```
Comment (danischm) [inline gen/definitions/evpn_segment_routing_service.yaml:45]: Let's remove the `evpn_` prefix here.

File: gen/definitions/evpn_segment_routing_service.yaml [hunk 19]
```diff
@@ -0,0 +1,81 @@
...
+    attributes:
+      - yang_name: two-byte-as-number
+        tf_name: as_number
+        id: true
+        example: 1
+      - yang_name: assigned-number
+        id: true
+        example: 1
+  - yang_name: bgp/route-target/export/four-byte-as-rts/four-byte-as-rt
+    tf_name: evpn_bgp_route_target_export_four_byte_as_format
```
Comment (danischm) [inline gen/definitions/evpn_segment_routing_service.yaml:57]: Let's remove the `evpn_` prefix here.

File: gen/definitions/segment_routing_v6.yaml (hunk 1 above)
Comment (swarudka) [reply gen/definitions/segment_routing_v6.yaml:22]: done

File: gen/definitions/segment_routing_v6.yaml (hunk 2 above)
Comment (swarudka) [reply gen/definitions/segment_routing_v6.yaml:24]: done

File: gen/definitions/segment_routing_isis.yaml (hunk 3 above)
Comment (swarudka) [reply gen/definitions/segment_routing_isis.yaml:1]: checking

Comment (swarudka) [reply gen/definitions/segment_routing_isis.yaml:1]: done

File: gen/definitions/segment_routing_te.yaml (hunk 4 above)
Comment (swarudka) [reply gen/definitions/segment_routing_te.yaml:2]: done

File: gen/definitions/segment_routing_te.yaml (hunk 5 above)
Comment (swarudka) [reply gen/definitions/segment_routing_te.yaml:16]: done

File: gen/definitions/segment_routing_te.yaml (hunk 6 above)
Comment (swarudka) [reply gen/definitions/segment_routing_te.yaml:22]: done

File: gen/definitions/segment_routing_te.yaml (hunk 7 above)
Comment (swarudka) [reply gen/definitions/segment_routing_te.yaml:24]: done

File: gen/definitions/segment_routing_te.yaml (hunk 9 above)
Comment (swarudka) [reply gen/definitions/segment_routing_te.yaml:31]: done

File: gen/definitions/segment_routing_te.yaml (hunk 10 above)
Comment (swarudka) [reply gen/definitions/segment_routing_te.yaml:49]: done

File: gen/definitions/segment_routing_te.yaml (hunk 11 above)
Comment (swarudka) [reply gen/definitions/segment_routing_te.yaml:55]: done

File: gen/definitions/segment_routing_te.yaml [hunk 20]
```diff
@@ -0,0 +1,64 @@
...
+        example: fccc:0:103::1
+      - yang_name: source-address/ip-address-type
```
[the remark this answers is not in the document]
Comment (swarudka) [reply gen/definitions/segment_routing_te.yaml:57]: done

File: gen/definitions/segment_routing_te.yaml [hunk 21]
```diff
@@ -0,0 +1,64 @@
...
+        example: end-point-type-ipv6
+      - yang_name: policy-color-endpoint/color
+        example: 65534
+      - yang_name: policy-color-endpoint/end-point-type
```
[the remark this answers is not in the document]
Comment (swarudka) [reply gen/definitions/segment_routing_te.yaml:61]: done

File: gen/definitions/segment_routing_te.yaml [hunk 22]
```diff
@@ -0,0 +1,64 @@
...
+        example: end-point-type-ipv6
+      - yang_name: policy-color-endpoint/end-point-address
```
[the remark this answers is not in the document]
Comment (swarudka) [reply gen/definitions/segment_routing_te.yaml:63]: done

File: gen/definitions/segment_routing_te_candidate_path.yaml [hunk 23]
```diff
@@ -0,0 +1,26 @@
+---
+name: Segment Routing TE Candidate Path
```
[the remark this answers is not in the document]
Comment (swarudka) [reply gen/definitions/segment_routing_te_candidate_path.yaml:2]: done

File: gen/definitions/segment_routing_te_candidate_path.yaml [hunk 24]
```diff
@@ -0,0 +1,26 @@
...
+  - yang_name: path-infos/path-info
+    tf_name: candidate_paths_type
+    type: List
+    attributes: 
+      - yang_name: type
+        id: true
+        example: dynamic
+      - yang_name: pcep
+        example: false
+      - yang_name: metric/metric-type
```
[the remark this answers is not in the document]
Comment (swarudka) [reply gen/definitions/segment_routing_te_candidate_path.yaml:19]: done

File: gen/definitions/segment_routing_te_candidate_path.yaml [hunk 25]
```diff
@@ -0,0 +1,26 @@
...
+name: Segment Routing TE Candidate Path
+path: Cisco-IOS-XR-segment-routing-ms-cfg:/sr/Cisco-IOS-XR-infra-xtc-agent-cfg:traffic-engineering/policies/policy[policy-name=%s]/candidate-paths/preferences/preference[path-index=%d]
+doc_category: Segment Routing
+attributes:
+  - yang_name: policy-name
+    example: POLICY1
+  - yang_name: path-index
+    example: 100
+  - yang_name: path-infos/path-info
+    tf_name: candidate_paths_type
```
[the remark this answers is not in the document]
Comment (swarudka) [reply gen/definitions/segment_routing_te_candidate_path.yaml:11]: done

File: gen/definitions/router_isis_address_family.yaml [hunk 26]
```diff
@@ -105,3 +105,21 @@ attributes:
         example: 1
       - yang_name: maximum-prefixes
         example: 1000
+
+  - yang_name: redistribute/isis
+    tf_name: redistribute_id
```
[the remark this answers is not in the document]
Comment (swarudka) [reply gen/definitions/router_isis_address_family.yaml:110]: This is nested list structure & we had followed the same way in bgp yaml file
example: router_bgp_address_family

File: gen/definitions/router_isis_address_family.yaml [hunk 27]
```diff
@@ -6,7 +6,7 @@ attributes:
   - yang_name: process-id
     example: P1
   - yang_name: af-name
-    example: ipv4
+    example: ipv6      #need to add support for ipv4 as well
```
[the remark this answers is not in the document]
Comment (swarudka) [reply gen/definitions/router_isis_address_family.yaml:9]: done

File: gen/definitions/router_isis_address_family.yaml (hunk 12 above)
Comment (swarudka) [reply gen/definitions/router_isis_address_family.yaml:120]: done

File: gen/definitions/router_bgp_neighbors_address_family.yaml [hunk 28]
```diff
@@ -0,0 +1,33 @@
+---
```
[the remark this answers is not in the document]
Comment (swarudka) [reply gen/definitions/router_bgp_neighbors_address_family.yaml:1]: the existing one is for VRF neighbor & i have created for BGP neighbors

Existing resource path: Cisco-IOS-XR-um-router-bgp-cfg:/router/bgp/as[as-number=%s]/vrfs/vrf[vrf-name=%s]/neighbors/neighbor[neighbor-address=%s]/address-families/address-family[af-name=%s]

New resource path: Cisco-IOS-XR-um-router-bgp-cfg:/router/bgp/as[as-number=%s]/neighbors/neighbor[neighbor-address=%s]/address-families/address-family[af-name=%s]

File: gen/definitions/l2vpn_xconnect_group_p2p.yaml (hunk 13 above)
Comment (swarudka) [reply gen/definitions/l2vpn_xconnect_group_p2p.yaml:22]: I was checking while running the test cases with different scenario, now its removed

File: gen/definitions/l2vpn_xconnect_group_p2p.yaml [hunk 29]
```diff
@@ -70,3 +71,16 @@ attributes:
         example: 1001
       - yang_name: pw-class
         example: PW_CLASS_1
+        
+  - yang_name: neighbor/evpn/evi/segment-routing-services/service
+    tf_name: evpn_segment_routing_service
```
[the remark this answers is not in the document]
Comment (swarudka) [reply gen/definitions/l2vpn_xconnect_group_p2p.yaml:76]: done

File: gen/definitions/l2vpn_bridge_group_bridge_domain.yaml (hunk 14 above)
Comment (swarudka) [reply gen/definitions/l2vpn_bridge_group_bridge_domain.yaml:56]: done

File: gen/definitions/evpn_segment_routing_service.yaml (hunk 15 above)
Comment (swarudka) [reply gen/definitions/evpn_segment_routing_service.yaml:2]: done

File: gen/definitions/evpn_segment_routing_service.yaml (hunk 16 above)
Comment (swarudka) [reply gen/definitions/evpn_segment_routing_service.yaml:22]: done

File: gen/definitions/evpn_segment_routing_service.yaml (hunk 17 above)
Comment (swarudka) [reply gen/definitions/evpn_segment_routing_service.yaml:34]: done

File: gen/definitions/evpn_segment_routing_service.yaml (hunk 18 above)
Comment (swarudka) [reply gen/definitions/evpn_segment_routing_service.yaml:45]: done

File: gen/definitions/evpn_segment_routing_service.yaml (hunk 19 above)
Comment (swarudka) [reply gen/definitions/evpn_segment_routing_service.yaml:57]: done

File: gen/definitions/evpn_segment_routing_service.yaml [hunk 30]
```diff
@@ -0,0 +1,81 @@
...
+    attributes:
+      - yang_name: four-byte-as-number
+        tf_name: as_number
+        id: true
+        example: 65536
+      - yang_name: assigned-number
+        id: true
+        example: 1
+  - yang_name: bgp/route-target/export/ipv4-address-rts/ipv4-address-rt
+    tf_name: evpn_bgp_route_target_export_ipv4_address_format
```
[the remark this answers is not in the document]
Comment (swarudka) [reply gen/definitions/evpn_segment_routing_service.yaml:69]: done

File: gen/definitions/evpn_segment_routing_service.yaml [hunk 31]
```diff
@@ -0,0 +1,81 @@
...
+    type: List
+    attributes:
+      - yang_name: ipv4-address
+        id: true
+        example: 1.1.1.1
+      - yang_name: assigned-number
+        id: true
+        example: 1
+  - yang_name: advertise-mac/bvi-mac
```
[the remark this answers is not in the document]
Comment (swarudka) [reply gen/definitions/evpn_segment_routing_service.yaml:78]: Corrected

File: CHANGELOG.md [hunk 32]
```diff
@@ -4,6 +4,12 @@
 - Add `iosxr_router_bgp_neighbor_group` resource and data source
 - Add `iosxr_router_static` resource and data source
 - Add `iosxr_key_chain` resource and data source
+- Add `iosxr_segment_routing_v6` resource and data source
```
[the remark this answers is not in the document]
Comment (swarudka) [reply CHANGELOG.md:7]: done

File: gen/definitions/l2vpn_xconnect_group_p2p.yaml (hunk 13 above)
Comment (swarudka) [reply gen/definitions/l2vpn_xconnect_group_p2p.yaml:22]: Reason: Set request failed, got error: rpc error: code = Internal desc = gNMI
        commit-request: operation-failed: Invalid argument: Cross-connect Segment
        combination not supported:
        Cisco-IOS-XR-um-l2vpn-cfg:l2vpn/xconnect/groups/group[group-name =
        'P2P']/p2ps/p2p[p2p-xconnect-name =
        'XC']/neighbor/evpn/evi/segment-routing-services/service[vpn-id = '4600' and
        service-id = '600']/vpn-id

**Review by danischm — APPROVED** [review]
